### PR TITLE
Warn about possible production overhead for the Go block profiler

### PR DESCRIPTION
**Pull request by nsrip-dd**
### What does this PR do?

Warn about possible production overhead for the Go block profiler

### Motivation

The Go block profiler can cause noticeable overhead, especially compared to other Go profile types, when enabled for production workloads.

### Preview

https://docs-staging.datadoghq.com/nick.ripley/go-block-profiler-warning/tracing/profiler/search_profiles/

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Review by maycmlee — CHANGES_REQUESTED** [review]

File: content/en/tracing/profiler/search_profiles.md [hunk 1]
```diff
@@ -158,7 +158,7 @@ Mutex
...
+: Shows the time functions have been waiting on mutexes and channel operations during the profiling period (default: 60s). Sleep, GC, Network and Syscall operations are not captured by this profile. Blocking operations are only captured after they become unblocked, so this profile cannot be used to debug applications that appear to be stuck. For mutex contentions, the stack traces in this profile point to blocked `Lock()` operations. This will tell you where your program is getting blocked, while the mutex profile tells you what part of your program is causing the contention. See our [Block Profiling in Go][1] research for more in-depth information. See also the note about how this measure changes in version `1.33.0` in [Delta profiles](#delta-profiles). *Note:* The block profiler can cause noticeable overhead for production workloads. If enabling it in production, prefer high rates (such as `100000000`, i.e. 100 milliseconds) and look for signs of increased latency or CPU utilization.
```
[the remark this answers is not in the document]
Comment (nsrip-dd) [reply content/en/tracing/profiler/search_profiles.md]: Thanks for the review!

**Review by maycmlee — APPROVED** [review]

File: content/en/tracing/profiler/search_profiles.md [hunk 2]
```diff
@@ -158,7 +158,9 @@ Mutex
...
+: Shows the time functions have been waiting on mutexes and channel operations during the profiling period (default: 60s). Sleep, GC, Network and Syscall operations are not captured by this profile. Blocking operations are only captured after they become unblocked, so this profile cannot be used to debug applications that appear to be stuck. For mutex contentions, the stack traces in this profile point to blocked `Lock()` operations. This will tell you where your program is getting blocked, while the mutex profile tells you what part of your program is causing the contention. See our [Block Profiling in Go][1] research for more in-depth information. See also the note about how this measure changes in version `1.33.0` in [Delta profiles](#delta-profiles). 
+
+**Note:** The block profiler can cause noticeable overhead for production workloads. If enabling it in production, prefer high rates (such as `100000000`, that is 100 milliseconds) and look for signs of increased latency or CPU utilization.
```
Comment (felixge) [inline content/en/tracing/profiler/search_profiles.md:163]: @maycmlee 👋 hey, we just noticed that this ended up breaking the box listing all the go profiles into two boxes which looks a bit odd. Was this intentional or should we send a PR to render the note text within the box?

image

Comment (maycmlee) [reply content/en/tracing/profiler/search_profiles.md:163]: Hey @felixge! Thanks for catching that! It's not intentional, I must have missed it in the preview. I'll open a PR to fix it.

Comment (maycmlee) [reply content/en/tracing/profiler/search_profiles.md:163]: PR: https://github.com/DataDog/documentation/pull/13211

Comment (felixge) [reply content/en/tracing/profiler/search_profiles.md:163]: Thank you so much for the fast fix : )

**Pull request merged**